test/functions.coffee

```diff
@@ -150,13 +150,13 @@ test "@-parameters and splats with constructors", ->
   eq b, obj.last
 
 test "destructuring in function definition", ->
-  (([{a: [b], c}]...) ->
+  (({a: [b], c}) ->
```

Comment:
So there was a period when I was struggling to get splatted arrays to work as function parameters, and so I modified these tests. Later on I added an [exception for this case](https://github.com/jashkenas/coffeescript/pull/4478/files#diff-d149b4f007169f06a8ad463f9b2f277bR2067) that just dumped such cases back into the legacy compilation, so now we could revert these test changes. But I think we shouldn’t, because the splat-array wrapper feels like a workaround. The version without the wrapper looks to me much more like something someone would intentionally write.